src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -0,0 +1,175 @@
+        $this->messageBus = $messageBus;
+        $this->publisher = $publisher;
+        $this->expressionLanguage = $expressionLanguage ?? class_exists(ExpressionLanguage::class) ? new ExpressionLanguage() : null;
+        $this->reset();
+    }
+
+    /**
+     * Collects created, updated and deleted entities.
+     */
+    public function onFlush(OnFlushEventArgs $eventArgs)
+    {
+        $uow = $eventArgs->getEntityManager()->getUnitOfWork();
+
+        foreach ($uow->getScheduledEntityInsertions() as $entity) {
+            $this->storeEntityToPublish($entity, 'createdEntities');
+        }
+
+        foreach ($uow->getScheduledEntityUpdates() as $entity) {
+            $this->storeEntityToPublish($entity, 'updatedEntities');
+        }
+
+        foreach ($uow->getScheduledEntityDeletions() as $entity) {
+            $this->storeEntityToPublish($entity, 'deletedEntities');
+        }
+    }
+
+    /**
+     * Purges tags collected during this request, and clears the tag list.
+     */
+    public function postFlush()
+    {
+        try {
+            foreach ($this->createdEntities as $entity) {
+                $this->publishUpdate($entity, $this->createdEntities[$entity]);
+            }
+
+            foreach ($this->updatedEntities as $entity) {
+                $this->publishUpdate($entity, $this->updatedEntities[$entity]);
+            }
+
+            foreach ($this->deletedEntities as $iri => $targets) {
+                $this->publishUpdate($iri, $targets);
+            }
+        } finally {
+            $this->reset();
+        }
+    }
+
+    private function reset(): void
+    {
+        $this->createdEntities = new \SplObjectStorage();
+        $this->updatedEntities = new \SplObjectStorage();
+        $this->deletedEntities = [];
+    }
+
+    /**
+     * @param object $entity
+     */
+    private function storeEntityToPublish($entity, string $property): void
+    {
+        $resourceClass = $this->getObjectClass($entity);
+        if (!$this->resourceClassResolver->isResourceClass($resourceClass)) {
+            return;
+        }
+
+        $value = $this->resourceMetadataFactory->create($resourceClass)->getAttribute('mercure', false);
+        if (false === $value) {
+            return;
+        }
+
+        if (\is_string($value)) {
+            if (null === $this->expressionLanguage) {
+                throw new RuntimeException('The Expression Language component is not installed. Try running "composer require symfony/expression-language".');
+            }
+
+            $value = $this->expressionLanguage->evaluate($value, ['object' => $entity]);
+        }
+
+        if (true === $value) {
+            $value = [];
+        }
+
+        if (!\is_array($value)) {
+            throw new InvalidArgumentException(sprintf('The value of the "mercure" attribute of the "%s" resource class must be a boolean, an array of targets or a valid expression, "%s" given.', $resourceClass, \gettype($value)));
```

Comment:
as above, proper place for this?

Reply:
We currently don't handle metadata validation in the factory, but it would definitely be a better place

Reply:
Yes. I'll keep it as is because it out of this PR's scope, and I'll try to add a proper validator (for all existing attributes) in a subsequent PR. Is it ok for you?

Reply:
of course! sorry, my `?!` wasn't meant to come off as rude or aggressive, but enthusiastic!

Reply:
I understood it like that no worries :)